tests/ha/single_pod_restart/test_server_pod_restart.py

```diff
@@ -0,0 +1,398 @@
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 3: Performed READs/Verify on data written in healthy cluster.")
+        LOGGER.info("Step 4: Perform WRITEs/READs/Verify with variable object sizes.")
+        self.test_prefix_deg = 'test-44834-deg'
+        resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
+                                                    log_prefix=self.test_prefix_deg,
+                                                    skipcleanup=True, setup_s3bench=False)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 4: Performed WRITEs/READs/Verify with variable sizes objects.")
+        LOGGER.info("Step 5: Restore server pod and check cluster status.")
+        resp = self.ha_obj.restore_pod(pod_obj=self.node_master_list[0],
+                                       restore_method=self.restore_method,
+                                       restore_params={"deployment_name": self.deployment_name,
+                                                       "deployment_backup":
+                                                           self.deployment_backup},
+                                       clstr_status=True)
+        LOGGER.debug("Response: %s", resp)
+        assert_utils.assert_true(resp[0], f"Failed to restore server pod by {self.restore_method} "
+                                          "way OR the cluster is not online")
+        LOGGER.info("Step 5: Successfully started server pod and cluster is online.")
+        self.restore_pod = False
+        LOGGER.info("Step 6: Perform READs and verify DI on the written data in degraded "
+                    "cluster with new buckets and objects.")
+        resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
+                                                    log_prefix=self.test_prefix_deg,
+                                                    skipwrite=True, skipcleanup=True,
+                                                    setup_s3bench=False)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 6: Successfully run READ/Verify on data written in degraded cluster "
+                    "with new buckets and objects.")
+        LOGGER.info("Step 7: Perform READs and verify DI on the written data with buckets created "
+                    "in healthy cluster.")
+        resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
+                                                    log_prefix=self.test_prefix, skipwrite=True,
+                                                    skipcleanup=True, setup_s3bench=False)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 7: Successfully run READ/Verify on data written with buckets created "
+                    "in healthy cluster")
+        LOGGER.info("Step 8: Start IOs again after server pod restart by delete deployment")
```

Comment:
we may not want to Write here as this is read only test

Reply:
This is IO in last step, so doing read/write/verify may be require as this is additional step

Reply:
We decided to do this in continues IOs tests scenarios only.

Reply:
no this is read only so we do not require this last step